src/main/java/com/example/bookstoreapp/repository/impl/BookRepositoryImpl.java

```diff
@@ -46,4 +47,16 @@ public List<Book> findAll() {
             throw new RuntimeException("Can`t get all Books", e);
         }
     }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            return Optional.ofNullable(entityManager
+                    .createQuery("SELECT b FROM Book b WHERE b.id = :id", Book.class)
+                    .setParameter("id", id)
+                    .getSingleResult());
```

Comment:
This method looks a little overcomplicated. It's up to you but there are some options to make it looks cleaner)

Reply:
> This method looks a little overcomplicated. It's up to you but there are some options to make it looks cleaner)

I don't like too many variables. I believe it is shortest possible solution. Might be wrong.

Reply:
you can simplify it a bit by using the method `find()` instead of a custom query